token/program-2022/src/extension/mod.rs

```diff
@@ -518,6 +518,26 @@ pub fn get_account_len(extension_types: &[ExtensionType]) -> usize {
     }
 }
 
+/// Get the list of required AccountType::Account ExtensionTypes based on a set of
+/// AccountType::Mint ExtensionTypes
+pub fn get_account_extensions(mint_extension_types: &[ExtensionType]) -> Vec<ExtensionType> {
+    let mut account_extension_types = vec![];
+    for extension_type in mint_extension_types {
+        #[allow(clippy::single_match)]
+        match extension_type {
+            ExtensionType::TransferFeeConfig => {
+                account_extension_types.push(ExtensionType::TransferFeeAmount);
+            }
+            #[cfg(test)]
+            ExtensionType::MintPaddingTest => {
+                account_extension_types.push(ExtensionType::AccountPaddingTest);
+            }
+            _ => {}
+        }
+    }
+    account_extension_types
+}
+
```

Comment:
nit: wdyt about moving this into `ExtensionType`? I did that for `get_account_len` and it felt cleaner

Reply:
I'm not opposed to that. What's the fn signature of your new `get_account_len`?

Reply:
Same as before: `pub fn get_account_len<S: BaseState>(extension_types: &[ExtensionType]) -> usize` Although now that I see it, should've done `Self` instead of `ExtensionType`.
https://github.com/solana-labs/solana-program-library/pull/2739/files#diff-1fd54e31043386c786ab6a3edf57cd653cf69082500a805e9dd50f550d2f260bR483-R505

Reply:
Yeah, it looks pretty nice. Will push.